File: apiwdbe/WdbeQBnkList.cpp
/**
	* \file WdbeQBnkList.cpp
	* API code for table TblWdbeQBnkList (implementation)
	* \author Alexander Wirthmueller
	* \date created: 11 Jul 2020
	* \date modified: 11 Jul 2020
	*/

#include "WdbeQBnkList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WdbeQBnkList
 ******************************************************************************/

WdbeQBnkList::WdbeQBnkList(
			const uint jnum
			, const string sref
			, const string stubRefWdbeMUnit
			, const string srefKVoltstd
			, const string titSrefKVoltstd
		) {
	this->jnum = jnum;
	this->sref = sref;
	this->stubRefWdbeMUnit = stubRefWdbeMUnit;
	this->srefKVoltstd = srefKVoltstd;
	this->titSrefKVoltstd = titSrefKVoltstd;
};

bool WdbeQBnkList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WdbeQBnkList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "stubRefWdbeMUnit", "unt", stubRefWdbeMUnit);
		extractStringUclc(docctx, basexpath, "srefKVoltstd", "vst", srefKVoltstd);
		extractStringUclc(docctx, basexpath, "titSrefKVoltstd", "vst2", titSrefKVoltstd);
	};

	return basefound;
};

/******************************************************************************
 class ListWdbeQBnkList
 ******************************************************************************/

ListWdbeQBnkList::ListWdbeQBnkList() {
};

ListWdbeQBnkList::ListWdbeQBnkList(
			const ListWdbeQBnkList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQBnkList(*(src.nodes[i]));
};

ListWdbeQBnkList& ListWdbeQBnkList::operator=(
			const ListWdbeQBnkList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQBnkList(*(src.nodes[i]));

	return *this;
};

ListWdbeQBnkList::~ListWdbeQBnkList() {
	clear();
};

void ListWdbeQBnkList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWdbeQBnkList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WdbeQBnkList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWdbeQBnkList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WdbeQBnkList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WdbeQBnkList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WdbeQBnkList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

File: japiwdbe/WdbeQBnkList.java
/**
	* \file WdbeQBnkList.h
	* Java API code for record of table TblWdbeQBnkList
	* \copyright (C) 2018-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

package apiwdbe;

import org.w3c.dom.*;
import sbecore.*;

public class WdbeQBnkList {
	
	public WdbeQBnkList(
				int jnum
				, String sref
				, String stubRefWdbeMUnit
				, String srefKVoltstd
				, String titSrefKVoltstd
			) {
		this.jnum = jnum;
		this.sref = sref;
		this.stubRefWdbeMUnit = stubRefWdbeMUnit;
		this.srefKVoltstd = srefKVoltstd;
		this.titSrefKVoltstd = titSrefKVoltstd;
	};

	public int jnum;
	public String sref;
	public String stubRefWdbeMUnit;
	public String srefKVoltstd;
	public String titSrefKVoltstd;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WdbeQBnkList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			stubRefWdbeMUnit = Xmlio.extractStringUclc(doc, basexpath, "stubRefWdbeMUnit", "unt", null, 0);
			srefKVoltstd = Xmlio.extractStringUclc(doc, basexpath, "srefKVoltstd", "vst", null, 0);
			titSrefKVoltstd = Xmlio.extractStringUclc(doc, basexpath, "titSrefKVoltstd", "vst2", null, 0);
			
			return true;
		};

		return false;
	};
};

